google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberImpl.java

```diff
@@ -115,13 +129,20 @@ protected void stop() {
 
   @Override
   public ApiFuture<Offset> seek(SeekRequest request) {
-    try (CloseableMonitor.Hold h = monitor.enter()) {
+    try (CloseableMonitor.Hold h =
+        monitor.enterWhenUninterruptibly(
+            new Monitor.Guard(monitor.monitor) {
+              @Override
+              public boolean isSatisfied() {
+                return !internalSeekInFlight;
```

Comment:
This should alternately check shutdown, or it can stall forever on a permanent failure.

Reply:
Good catch - thanks! Fixed.